### PR TITLE
Implement dashboard management

### PR DESCRIPTION
**Pull request by leifmadsen**
Implement a new configuration option graphing.grafana.dashboards.enabled
which results in dashboards objects being created for the Grafana
Operator. Previously loading dashboards would be done manually via 'oc
apply' using instructions from documentation.

The new CRD parameters to the ServiceTelemetry object allows the Service
Telemetry Operator to now make the GrafanaDashboard objects directly.

Related: OSPRH-825

### CODE REVIEW

File: deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml [hunk 1]
```diff
@@ -79,10 +79,13 @@ spec:
   graphing:
     enabled: false
     grafana:
-      ingressEnabled: false
+      ingressEnabled: true
```
Comment (leifmadsen) [inline deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml:82]: I figured if you go to the trouble of enabling dashboarding you probably want to access it.

File: deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml [hunk 2]
```diff
@@ -79,10 +79,13 @@ spec:
...
       adminPassword: secret
       adminUser: root
       disableSignoutMenu: false
+      baseImage: registry.redhat.io/rhel8/grafana:7
```
Comment (leifmadsen) [inline deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml:86]: Using this as an opportunity to update our base image to use something closer to being supported, and with CVE update tracking.

File: deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml [hunk 3]
```diff
@@ -79,10 +79,13 @@ spec:
   graphing:
     enabled: false
     grafana:
-      ingressEnabled: false
+      ingressEnabled: true
       adminPassword: secret
       adminUser: root
```
Comment (leifmadsen) [inline deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml:84]: These will eventually go away in another PR.

File: deploy/olm-catalog/service-telemetry-operator/manifests/service-telemetry-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -175,32 +179,8 @@ spec:
...
-        version: v1
-      - kind: Deployments
-        name: ""
-        version: v1
-      - kind: Services
-        name: ""
-        version: v1
-      - kind: ServiceMonitors
-        name: servicemonitors.monitoring.coreos.com
-        version: v1
```
Comment (leifmadsen) [inline deploy/olm-catalog/service-telemetry-operator/manifests/service-telemetry-operator.clusterserviceversion.yaml:203]: I was trying to figure out why `operator-sdk generate bundle [...]` stripped all these resources out (since the docs seem to imply this isn't a managed interface?). I was going through to figure out why this was the case, and all I can think of is that because I moved some resources up to the cluster scope level in the roles.yaml that it wanted to strip these out as not being self-managed?

File: deploy/olm-catalog/service-telemetry-operator/manifests/service-telemetry-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -211,6 +191,18 @@ spec:
     spec:
       clusterPermissions:
       - rules:
+        - apiGroups:
+          - apiextensions.k8s.io
+          resourceNames:
+          - servicetelemetrys.infra.watch
```
Comment (leifmadsen) [inline deploy/olm-catalog/service-telemetry-operator/manifests/service-telemetry-operator.clusterserviceversion.yaml:197]: I actually think I can drop this to be honest. I was using some previous work as the kick off, and I have a feeling this was here to allow reading the CRD as part of some debugging. I'll go through everything again and see if this can be removed. If so, then I'll try to put back the other `resources` list in the CSV, although I'm not really sure if _that_ is necessary.

I noticed that it had ServiceMonitors for the old CRD path when using Prometheus Operator (coreos.com) vs the COO one (rhobs). Since everything seems to work, as far as I can tell this is just informational, and we can probably just leave it in the simplified form?

Comment (leifmadsen) [reply deploy/olm-catalog/service-telemetry-operator/manifests/service-telemetry-operator.clusterserviceversion.yaml:197]: Actually I think this whole block might have been for debugging purposes, so there is a good chance this whole thing will go away.

File: roles/servicetelemetry/tasks/component_grafana.yml [hunk 6]
```diff
@@ -115,3 +115,74 @@
         state: '{{ "present" if servicetelemetry_vars.graphing.enabled else "absent" }}'
         definition:
           '{{ ds_manifest }}'
+
+    - name: Load Cloud Overview Dashboard
+      k8s:
+        state: '{{ "present" if servicetelemetry_vars.graphing.grafana.dashboards.enabled else "absent" }}'
```
Comment (leifmadsen) [inline roles/servicetelemetry/tasks/component_grafana.yml:121]: TODO: test that disabling dashboards will also remove them (and that I have set the appropriate RBAC rules to allow for that).

File: roles/servicetelemetry/tasks/component_grafana.yml [hunk 7]
```diff
@@ -115,3 +115,74 @@
...
+            labels:
+              app: grafana
+              stf_owner: "{{ ansible_operator_meta.name }}"
+            name: rhos-cloud-dashboard-1
+            namespace: "{{ ansible_operator_meta.namespace }}"
+          spec:
+            name: rhos-cloud-dashboard.json
+            plugins:
+              - name: grafana-polystat-panel
+                version: "1.2.11"
```
Comment (leifmadsen) [inline roles/servicetelemetry/tasks/component_grafana.yml:135]: I thought about being fancy with a loop or whatever here and making this somewhat dynamic, but then I re-read the initial comment in https://issues.redhat.com/browse/STF-1623 which says, "Make sure not to overly complicate the management of the dashboards if at all possible."

None of the other dashboards need this plugin so I _could_ probably simplify this code path a bit with a loop, and maybe even some extra args or whatever with a loop over a hash, but I figured for the 4 dashboards here, it's just as easy to break them out and show them all.

File: deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml (hunk 1 above)
Comment (leifmadsen) [reply deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml:82]: I think this was our default because we wanted to limit the attack surface but in this case, if you want dashboarding, you're probably fine with exposing the route :)

Comment (leifmadsen) [reply deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml:82]: (heh I was curious and ran a git blame, and the defaults/main.yml for this setting has been there since STF 1.1.0, so I don't think there was anything as smart as limiting attack surface... it was probably just the default value that Grafana Operator was using, and we likely just copied it? 🤷🏻 )

File: deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml (hunk 3 above)
Comment (leifmadsen) [reply deploy/crds/infra.watch_v1beta1_servicetelemetry_cr.yaml:84]: #550

File: roles/servicetelemetry/tasks/component_grafana.yml (hunk 6 above)
Comment (leifmadsen) [reply roles/servicetelemetry/tasks/component_grafana.yml:121]: ✅ works as intended

**Comment by leifmadsen** [comment]
Ready for review now

**Comment by elfiesmelfie** [comment]
> Build failed (check pipeline). Post `recheck` (without leading slash) to rerun all jobs. Make sure the failure cause has been resolved before you rerun jobs.
> 
> https://review.rdoproject.org/zuul/buildset/7ee2cebc490d4ce5b1793591796196f8
> 
> ✔️ [stf-crc-ocp_412-nightly_bundles ](https://review.rdoproject.org/zuul/build/18fc4c09813a4d7eb23353b289936802) SUCCESS in 26m 24s ✔️ [stf-crc-ocp_412-local_build ](https://review.rdoproject.org/zuul/build/03efd31c32a44dcebe5b5042f656611e) SUCCESS in 38m 15s ✔️ [stf-crc-ocp_412-local_build-index_deploy ](https://review.rdoproject.org/zuul/build/56e622c7dd904e90a145b4195e7648d4) SUCCESS in 40m 15s ✔️ [stf-crc-ocp_413-nightly_bundles ](https://review.rdoproject.org/zuul/build/603c3171a2d14cae919dbb31ac78cc05) SUCCESS in 26m 03s ✔️ [stf-crc-ocp_413-local_build ](https://review.rdoproject.org/zuul/build/ff1e1b923e534baf9d2e2d6005c72679) SUCCESS in 37m 57s ❌ [stf-crc-ocp_413-local_build-index_deploy ](https://review.rdoproject.org/zuul/build/adb4e00357d54ea18b982369728c8665) **FAILURE** in 34m 35s

The failure is unrelated to this change.

**Review by csibbitt — APPROVED** [review]
